pkg/lint/example/example.go

```diff
@@ -0,0 +1,132 @@
+// Package example provides example linters to demonstrate the linter framework.
+// These linters are for demonstration purposes and are not registered by default.
+package example
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/block/spirit/pkg/lint"
+	"github.com/block/spirit/pkg/statement"
+)
+
+// TableNameLengthLinter checks that table names are not too long.
+// MySQL has a limit of 64 characters for table names.
+type TableNameLengthLinter struct {
+	maxLength int
+}
+
+// TableNameLengthConfig holds configuration for the table name length linter.
+type TableNameLengthConfig struct {
+	MaxLength int `json:"max_length"`
+}
+
+// NewTableNameLengthLinter creates a new table name length linter with default configuration.
+func NewTableNameLengthLinter() *TableNameLengthLinter {
+	return &TableNameLengthLinter{
+		maxLength: 64, // MySQL's limit
```

Comment:
Nit: because we need to spirit migrate, we should limit at ~58 characters. There is an existing check in checks/ which checks this too.